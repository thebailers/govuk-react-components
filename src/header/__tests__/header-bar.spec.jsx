import React from 'react';
import { render } from 'enzyme';
import Header from 'govuk-react-components/header/header-bar';

describe('The Header Bar Component', () => {
  it('should render correctly', () => {
    expect(
      render(<Header propositionHeader='Lorem ipsum' />)
    ).toMatchSnapshot();
  });

  it('should render correctly when no propositionHeader is passed', () => {
    expect(render(<Header />)).toMatchSnapshot();
  });

  it('should render a link when passed', () => {
    expect(
      render(<Header propositionHeader="Lorem ipsum" propositionHeaderLink="/" />)
    ).toMatchSnapshot();
  });
});
