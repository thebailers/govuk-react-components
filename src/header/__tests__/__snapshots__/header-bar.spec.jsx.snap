// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Header Bar Component should render a link when passed 1`] = `
<header
  class="with-proposition"
  id="global-header"
  role="banner"
>
  <div
    class="header-wrapper"
  >
    <div
      class="header-global"
    >
      <div
        class="header-logo"
      >
        <a
          class="content"
          href="https://www.gov.uk"
          id="logo"
          title=""
        >
          <img
            alt=""
            height="32"
            src="/govuk/images/gov.uk_logotype_crown_invert_trans.png?0.23.0"
            width="36"
          />
           GOV.UK
        </a>
      </div>
    </div>
    <div
      class="header-proposition"
    >
      <div
        class="content"
      >
        <nav
          id="proposition-menu"
        >
          <a
            href="/"
            id="proposition-name"
          >
            Lorem ipsum
          </a>
          <ul
            id="proposition-links"
          >
            <li>
              <a
                class="active"
                href="url-to-page-1"
              >
                Navigation item #1
              </a>
            </li>
            <li>
              <a
                class=""
                href="url-to-page-2"
              >
                Navigation item #2
              </a>
            </li>
            <li>
              <a
                class=""
                href="url-to-page-3"
              >
                Navigation item #3
              </a>
            </li>
          </ul>
        </nav>
      </div>
    </div>
  </div>
</header>
`;

exports[`The Header Bar Component should render correctly 1`] = `
<header
  class="with-proposition"
  id="global-header"
  role="banner"
>
  <div
    class="header-wrapper"
  >
    <div
      class="header-global"
    >
      <div
        class="header-logo"
      >
        <a
          class="content"
          href="https://www.gov.uk"
          id="logo"
          title=""
        >
          <img
            alt=""
            height="32"
            src="/govuk/images/gov.uk_logotype_crown_invert_trans.png?0.23.0"
            width="36"
          />
           GOV.UK
        </a>
      </div>
    </div>
    <div
      class="header-proposition"
    >
      <div
        class="content"
      >
        <nav
          id="proposition-menu"
        >
          <a
            href="/"
            id="proposition-name"
          >
            Lorem ipsum
          </a>
          <ul
            id="proposition-links"
          >
            <li>
              <a
                class="active"
                href="url-to-page-1"
              >
                Navigation item #1
              </a>
            </li>
            <li>
              <a
                class=""
                href="url-to-page-2"
              >
                Navigation item #2
              </a>
            </li>
            <li>
              <a
                class=""
                href="url-to-page-3"
              >
                Navigation item #3
              </a>
            </li>
          </ul>
        </nav>
      </div>
    </div>
  </div>
</header>
`;

exports[`The Header Bar Component should render correctly when no propositionHeader is passed 1`] = `
<header
  class=""
  id="global-header"
  role="banner"
>
  <div
    class="header-wrapper"
  >
    <div
      class="header-global"
    >
      <div
        class="header-logo"
      >
        <a
          class="content"
          href="https://www.gov.uk"
          id="logo"
          title=""
        >
          <img
            alt=""
            height="32"
            src="/govuk/images/gov.uk_logotype_crown_invert_trans.png?0.23.0"
            width="36"
          />
           GOV.UK
        </a>
      </div>
    </div>
  </div>
</header>
`;
